CHANGES.rst

```diff
@@ -27,6 +27,19 @@ Olivier Grisel
     ``prefer='threads'`` or enforce shared-memory semantics with
     ``require='sharedmem'``.
 
+Alexandre Abadie
+
+    Introduce the concept of 'store' and refactor the ``Memory`` internal
+    storage implementation to make it accept extra store backends for caching
+    results. ``backend`` and ``backend_options`` are the new options added to
+    ``Memory`` to specify and configure a store backend.
+
+    Add the ``register_store_backend`` function to extend the store backend
+    used by default with Memory. This default store backend is named 'local'
+    and corresponds to the local filesystem.
```

Comment:
Maybe add a note to say that the store backend API itself is experimental and subject to change without deprecation.

Reply:
ok

Reply:
note added